examples/async-universal/common/api/reddit.js

```diff
@@ -0,0 +1,14 @@
+import { fetchPostsIfNeeded, selectReddit } from '../actions'
+
+export function fetchData(store, subreddit, callback) {
+
+  if (!subreddit) return callback()
```

Comment:
I don't think `api` is a good folder name for this by the way. It's just a leftover from earlier example.

My intention with this example is for it to *also* have routing. 

I would like to:

* Add routing here 
* Move this function to the container component
* Add code to gather all `fetchData()` from matching routes before rendering instead of hard coding a particular call

Thanks!